plugins/rest/auth.go

```diff
@@ -340,6 +341,14 @@ func (ap *oauth2ClientCredentialsAuthPlugin) requestToken(ctx context.Context) (
 			if ap.ClientID != "" {
 				body.Add("client_id", ap.ClientID)
 			}
+		} else if ap.ClientAssertionFile != "" {
+			body.Add("client_assertion_type", "urn:ietf:params:oauth:client-assertion-type:jwt-bearer")
```

Comment:
We should probably add a `ClientAssertionType` field to the plugin struct and default it to `urn:ietf:params:oauth:client-assertion-type:jwt-bearer`. Also a `ClientAssertion` field to set the value directly and not a file path would be helpful too.

Reply:
I will push an update

Reply:
I will push an update

Reply:
@ashutosh-narkar  could you point me which test case is still failing?  I couldn't find the failed test method